File: test69_out/test69/Test_12/Test_12.js

/*
 * ======== MODULE test69.Test_12 ========
 */

var $$c = function() {
    const Test_12 = {};
    Test_12.Test_12 = Test_12
    Test_12.$name = 'test69.Test_12';
    Test_12.pollen$used = 0;
    Test_12.Events = $units['pollen.event.Events'];
    Test_12.CP = $units['virtual.mcu.ConsolePrint'];
    Test_12.PrintImpl = $units['test54.PrintImpl'];
    Test_12.pollen__run = new $$Ref('test69_Test_12_pollen__run__E');
    Test_12.tick = new $$Ref('test69_Test_12_tick__I');
    Test_12.pollen__shutdown = new $$Ref('test69_Test_12_pollen__shutdown__E');
    Test_12.targetInit = new $$Ref('test69_Test_12_targetInit__I');
    Test_12.$$hostInit = function() {
        var $$text = '';
        Test_12.e = new $units['pollen.event.Events'].Event('test69_Test_12_e__V');
        (Test_12.e) = new $units['pollen.event.Events'].Event().new_host((Test_12.tick));
        return $$text;
    }
    Test_12.pollen__uses__ = function() {
        $$bind($units['pollen.event.Events'], 'pollen$used', true);
        $$bind($units['virtual.mcu.ConsolePrint'], 'pollen$used', true);
        $$bind($units['test54.PrintImpl'], 'pollen$used', true);
    }
    Test_12.$$privateInit = function() {
var $$v = 'test54_PrintImpl';
        var $$s = $$bind($units['test69.Test_12'], 'pollenPrintProxy', $$v);{ $$v.pollen$used = true;
        $units['test54.PrintImpl'].pollen$used = true;}

    }
    return Test_12;
}
var $$u = $$c();
$units['test69.Test_12'] = $$u;
$units.push($$u);

File: test69_out/test69/Test_12/Test_12.c

/*
 * ======== string literals ========
 */

static const char test69_Test_12__s__fileName[] = "Test_12.p";
static const char test69_Test_12__s__0[] = "TICK TICK TOCK!\n";
static const char test69_Test_12__s__2[] = "Test_12";
static const char test69_Test_12__s__1[] = "SHUTDOWN\n";

/*
 * ======== function definitions ========
 */

void test69_Test_12_pollen__run__E() {
    pollen_event_Events_test__E(&(test69_Test_12_e__V));
}

void test69_Test_12_tick__I() {
    virtual_mcu_ConsolePrint_printStr__E(test69_Test_12__s__0);
}

void test69_Test_12_pollen__shutdown__E( uint8 id ) {
    virtual_mcu_ConsolePrint_printStr__E(test69_Test_12__s__1);
}

void test69_Test_12_targetInit__I() {
}

